### ISSUE TITLE
Getting strange shadow when click a menu button which doesn't has any button.

### ISSUE THREAD

**Issue by willser** [emilk/egui#1537]
Hi ,I have a strange bug writing my new project by using egui.

**Describe the bug**
Getting strange shadow when click a menu button which doesn't has any button.

**To Reproduce**
```rust
egui::menu::bar(ui, |ui| {
    ui.menu_button("New", |ui| {
        if ui.button("Http").clicked() {
            self.requests.insert(0, Http::default());
        }
    });
    ui.menu_button("Settings", |_ui| self.settings.show_settings = true);
});
```
In [weaver](https://github.com/willser/weaver/blob/9e097f3950606746a007353f3ec871a46776f065/src/main.rs#L50-L64)

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/64425242/165519904-d80eb465-7c7b-45d6-91c6-57d3d80ad3db.png)

It's ok if the menu button has a button.
![](https://user-images.githubusercontent.com/64425242/165519489-84f98191-51ac-48f0-96c2-454961ef4385.PNG)

**Desktop (please complete the following information):**
 - OS: `windows 10`
 - Version `0.17.0`

Thank you for your contribution,this is a a amazing crate.

**Comment by emilk** [comment]
There's not much sense in opening an empty menu, and I guess the painting code doesn't handle it well. Seems like you would want to use a normal button in this case.

**Comment by willser** [comment]
> There's not much sense in opening an empty menu, and I guess the painting code doesn't handle it well. Seems like you would want to use a normal button in this case.

Yeah.You are right.It's not common that there is an empty menu.Maybe I should add a button below it.Thank you.:smiley:
